Replace mmCIF rsync FTP download with AWS CLI download from Amazon S3 PDB Snaphots

teodorpopescuQB opened the pull request:
This PR contains:
- An additional requirement to have pip dependency manager installed for the MsaServer setup
- Changes that speed up the mmCIF database download, replacing the FTP server download via rsync with an S3 download of the mmcif snapshot from the [Registry of Open Data on AWS](https://registry.opendata.aws/pdb-3d-structural-biology-data) via the AWS CLI

milot-mirdita commented:
I would prefer not to replace it completely but to keep it as an option of either rsync or aws.

Also younger checking for the presence of pip instead of aws.

teodorpopescuQB commented:
I have now adjusted the code to keep the aws download optional based on a variable.
I didn't get the younger checking comment but I assumed you meant checking for aws instead of pip, so I replaced that.

milot-mirdita commented:
I didn't actually notice that pip was included in the other `setup-and-start-local` script. I don't think the check for `pip` or `aws` makes sense there.

I moved the check to the `setup_databases` script. I think it also makes sense to sync the latest changes through rsync after downloading the initial batch of structures through aws.

I didn't find a way to always use the latest snapshot or the s3.rcsb.org site. So this will be a yearly annoyance I guess :/
